watchtower-plugin/src/main.rs

```diff
@@ -133,15 +135,24 @@ async fn get_registration_receipt(
     plugin: Plugin<Arc<Mutex<WTClient>>>,
     v: serde_json::Value,
 ) -> Result<serde_json::Value, Error> {
-    let tower_id = TowerId::try_from(v).map_err(|x| anyhow!(x))?;
+    let params = GetRegistrationReceiptParams::try_from(v).map_err(|x| anyhow!(x))?;
+    let tower_id = params.tower_id;
+    let subscription_start = params.subscription_start;
+    let subscription_expiry = params.subscription_expiry;
     let state = plugin.state().lock().unwrap();
 
-    if let Some(response) = state.get_registration_receipt(tower_id) {
-        Ok(json!(response))
+    let response =
+        state.get_registration_receipt(tower_id, subscription_start, subscription_expiry);
+    if response.clone().unwrap().is_empty() {
+        if state.towers.contains_key(&tower_id) {
+            Err(anyhow!("No registration receipt found for {tower_id}"))
```

Comment:
Given this implies that the tower is found, an it cannot be the case that a tower is found and a registration receipt is not, it means it can only be hit if a boundary is passed and there is no receipt for that boundary. Therefore: 
```suggestion
            Err(anyhow!("No registration receipt found for {tower_id} on the given range"))
```